### DISCUSSION TITLE
The reviewer rules

### DISCUSSION THREAD

**Discussion by ming-tsai** [MakeContributions/DSA#232] [General]
In this discussion, we are going to talk about the rules of being a reviewer.
1. The recent change of file should be the reviewer.
2. Don't review what another reviewer has already reviewed, at least you find a mistake or other reviewer might have missed
3. Do the changes mentioned by @...
4. **DO NOT** merge the pull request that is already accepted.

Please add your suggetions:
@Arsenic-ATG, @UG-SEP, @aayushjain7, @TawfikYasser, @cyberwizard1001, @atin, @paulsonjpaul, @sridhar-5

**Comment by Arsenic-ATG** [comment]
> 2. Do not review if someone is reviewing, at least the reviewer approved the pull request but it has a better way to code

I think instead of this, it should be " don't review what other reviewers have already reviewed. ". I don't see a problem if someone else find a mistake which other reviewer might have missed ( after all we are humans ). More reviewer will only mean more checks before getting the changes accepted, but if all of them point out same thing or just say " do the changes mentioned by @... " then it can become a mess.

**Reply by ming-tsai** [reply]
I have updated

**Reply by ming-tsai** [reply]
The second suggestion, I mean if someone approved it, another one review the pull request and doesn't find out any problem, you don't have approved it

**Comment by TawfikYasser** [comment]
Hi @ming-tsai  I'm totally agree with you.

**Comment by cyberwizard1001** [comment]
Hi, @ming-tsai, I feel suggestions to improvement are always welcome irrespective of when its being given. I agree with @Arsenic-ATG , suggestions like "Please implement what @x said" can be avoided, genuine comments can be welcome.

**Comment by UG-SEP** [comment]
@ming-tsai  @Arsenic-ATG  @cyberwizard1001  @TawfikYasser Yes, You all are right.
If I have done any mistake then please forgive me i will follow all the rules from now

**Comment by aayushjainx** [comment]
Agreed! Will follow rules from now. 💯

**Comment by ming-tsai** [comment]
Hi @Waqar-107,
Do you want to be a C# reviewer?

**Reply by Waqar-107** [reply]
@ming-tsai sure!
